internal/keepers/service.go

```diff
@@ -89,12 +89,18 @@ func (s *onDemandUpkeepService) CheckUpkeep(ctx context.Context, key types.Upkee
 	if err != nil {
 		return result, err
 	}
-	state, stateCached := s.stateCache.Get(string(id))
 
+	// if an upkeep is in-flight (reported but not confirmed) all checks should
+	// return the InFlight status
+	_, isInFlight := s.inFlightCache.Get(string(id))
```

Comment:
wdyt about moving this logic outside checkUpkeep (to the plugin)? 

Would limit the purpose of this function to checkUpkeep using checkCache, while inFlightCache remains a plugin concept

Reply:
The problem I see with this is that the service needs to know which upkeeps to return a cached result for. Maybe inflight upkeeps could be passed to `SampleUpkeeps`? That would separate concerns a bit. wdyt?

Reply:
yep I think that would make sense. Let's discuss more offline